libs/pika/execution/include/pika/execution/algorithms/drop_operation_state.hpp

```diff
@@ -248,5 +249,14 @@ namespace pika::execution::experimental {
         {
             return detail::partial_algorithm<drop_operation_state_t>{};
         }
-    } drop_operation_state{};
+    };
+
+    /// \brief Release any operation state coming before the adaptor before it signals a connected
+    ///  receiver.
+    ///
+    /// Sender adaptor that takes any sender, release any operation state coming before this adaptor
+    /// and returns a new sender.
```

Comment:
If one knows something about operation states one knows the following can't happen, but with the current wording this seems to imply an order:

- take any sender
- release the operation state
- return a new sender

which is not true (since there's no operation state yet at the point when the new sender is returned).

Perhaps you can separately describe what the sender adaptor does to completion signatures etc. (i.e. _almost_ nothing, except change lvalue references to rvalue references), and then separately describe what the adaptor does when it's signalled and before it signals a receiver?